File: flutter_application_1/kec-study-hub-backend/models/Report.js
const mongoose = require('mongoose');

const reportSchema = new mongoose.Schema({
  materialId: { type: mongoose.Schema.Types.ObjectId, ref: 'StudyMaterial', required: true },
  reportedBy: { type: String, required: true },
  reason: { type: String, required: true },
  reportedAt: { type: Date, default: Date.now },
});

module.exports = mongoose.model('Report', reportSchema);

File: flutter_application_1/lib/models/report.dart
class Report {
  final String id;
  final String materialId;
  final String reportedBy;
  final String reason;
  final DateTime reportedAt;

  Report({required this.id, required this.materialId, required this.reportedBy, required this.reason, required this.reportedAt});

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['_id'],
      materialId: json['materialId'],
      reportedBy: json['reportedBy'],
      reason: json['reason'],
      reportedAt: DateTime.parse(json['reportedAt']),
    );
  }
}